R/predict.R

```diff
@@ -0,0 +1,512 @@
+  newdata <- create_newdata_dataframe(
+    object = object,
+    parameter = "r",
+    mean_delay = mean_delay,
+    min_date = min_date,
+    max_date = max_date,
+    horizon = horizon,
+    delta = delta
+  )
+  fitted <- gratia::fitted_samples(
+    object[["model"]],
+    data = newdata,
+    n = n,
+    seed = seed,
+    unconditional = TRUE,
+    scale = "linear_predictor",
+    ...
+  )
+  format_predicted_dataframe(
+    parameter = "r",
+    fitted = fitted,
+    newdata = newdata,
+    delta = delta
+  )
+}
+
+#' @rdname predictor
+#' @export
+predict_rt <- function(
+    object,
+    horizon = NULL,
+    min_date = NULL,
+    max_date = NULL,
+    n = 10,
+    gi_pmf = NULL,
+    seed = 12345,
+    mean_delay,
+    call = rlang::caller_env(),
+    ...) {
+  newdata <- create_newdata_dataframe(
+    object = object,
+    parameter = "Rt",
+    mean_delay = mean_delay,
+    min_date = min_date,
+    max_date = max_date,
+    horizon = horizon,
+    gi_pmf = gi_pmf
+  )
+  fitted <- gratia::fitted_samples(
+    object[["model"]],
+    data = newdata,
+    n = n,
+    seed = seed,
+    unconditional = TRUE,
+    scale = "response",
+    ...
+  )
+  format_predicted_dataframe(
+    parameter = "Rt",
+    fitted = fitted,
+    newdata = newdata,
+    gi_pmf = gi_pmf
+  )
+}
+
+#' Convert from user specification to necessary date range
+#'
+#' @inheritParams predict.RtGam
+#' @param call The calling environment to be reflected in the error message
+#'
+#' @return List with two elements: min_date and max_date
+#' @keywords internal
+#' @importFrom rlang %||%
+parse_predict_dates <- function(
+    object,
+    min_date = NULL,
+    max_date = NULL,
+    horizon = NULL,
+    call = rlang::caller_env()) {
+  if (!rlang::is_null(min_date)) check_date(min_date, call = call)
+  if (!rlang::is_null(max_date)) check_date(max_date, call = call)
+  if (!rlang::is_null(horizon)) check_integer(horizon, call = call)
+
+  # Handle horizon to estimate dates if provided
+  if (!rlang::is_null(horizon)) {
+    if (!rlang::is_null(max_date)) {
+      cli::cli_abort("Cannot specify both {.arg horizon} and {.arg max_date}",
+        call = call
+      )
+    }
+    min_date <- min_date %||% (object[["max_date"]] + 1)
+    max_date <- object[["max_date"]] + horizon
+  } else {
+    # Default to object's date range if not specified
+    min_date <- min_date %||% object[["min_date"]]
+    max_date <- max_date %||% object[["max_date"]]
+  }
+
+  # Ensure min_date is before max_date
+  if (min_date >= max_date) {
+    cli::cli_alert_warning("Swapping {.arg min_date} and {.arg max_date}")
+    cli::cli_alert(c(
+      "{.arg min_date} {.val {min_date}} ",
+      "is after {.arg max_date} {.val {max_date}}"
+    ))
+    temp_var <- max_date
+    max_date <- min_date
+    min_date <- temp_var
+  }
+
+  seq.Date(
+    from = min_date,
+    to = max_date,
+    by = "day"
+  )
+}
+
+#' Map user-req'd dates to model-applied dates. Incidence is a
```

Comment:
I hadn't understoond what you meant by incidence until this doc string. I am not sure this is a great name for it honestly. What about shifted cases or latent cases or similar?

Reply:
Dropped `obs_incidence` https://github.com/CDCgov/cfa-gam-rt/pull/48/commits/1836493540a3f726c06ce6b82cf0d1322d6ffa48